zeq/connection/broker.h

```diff
@@ -0,0 +1,56 @@
+
+/* Copyright (c) 2014-2015, Human Brain Project
+ *                          Stefan.Eilemann@epfl.ch
+ */
+
+#ifndef ZEQ_CONNECTION_BROKER_H
+#define ZEQ_CONNECTION_BROKER_H
+
+#include <zeq/receiver.h> // base class
+#include <lunchbox/debug.h>
+
+namespace zeq
+{
+namespace connection
+{
+namespace detail { class Broker; }
+
+/**
+ * Brokers subscription requests for a zeq::Receiver.
+ *
+ * Example: @include tests/connection/broker.cpp
+ */
+class Broker : public Receiver
+{
+public:
+    /**
+     * Create a new subscription broker.
+     *
+     * The given receiver has to have at least the same lifetime as this
+     * broker. The receiver and broker are automatically shared.
+     *
+     * For simplicity, only a single Receiver is handled by a Broker. The
+     * implementation should be extended if multiple receivers shall be
+     * handled.
+     *
+     * @param address the zmq reply socket address to be used.
+     * @param receiver the Receiver to manage.
+     * @throw std::runtime_error when the zmq setup failed.
+     */
+    ZEQ_API Broker( const std::string& address, Receiver& receiver );
```

Comment:
> why the broker and the subscriber need to be bound together 

Because: 

> broker is [...] responsible of introducing publishers to subscribers

and therefore needs to know the subscribers. It is a connection::Broker, and I don't see how it can be made any more complex than it is (in terms of API), or why it should be? It is a first minimal functional implementation and as such can evolve forward and potentially split into different impls later.